examples/SCAMPy/src/helper_funcs.jl

```diff
@@ -109,12 +113,85 @@ function run_SCAMPy(u::Array{FT, 1},
         end
         println("LENGTH OF G_SCM_ARR : ", length(g_scm))
         println("LENGTH OF G_SCM_ARR_PCA : ", length(g_scm_pca))
-        return g_scm, g_scm_pca
+        return sim_dirs, g_scm, g_scm_pca
     else
-        return g_scm
+        return sim_dirs, g_scm
     end
 end
 
+
+"""
+    function run_SCAMPy_handler(
+        u::Array{FT, 1},  
+        u_names::Array{String, 1},
+        scampy_dir::String,
+        scm_names::String,
+        scm_data_root::String,
+    ) where {FT<:AbstractFloat}
+
+Run a list of cases using a set of parameters `u_names` with values `u`,
+and return a list of directories pointing to where data is stored for 
+each simulation run.
+
+Inputs:
+ - u :: Values of parameters to be used in simulations.
+ - u_names :: SCAMPy names for parameters u.
+ - scampy_dir :: Path to SCAMPy directory
+ - scm_names :: Names of SCAMPy cases to run
+ - scm_data_root :: Path to SCAMPy case data (<scm_data_root>/Output.<scm_name>.00000)
+Outputs:
+ - output_dirs :: list of directories containing output data from the SCAMPy runs.
+"""
+function run_SCAMPy_handler(
+        u::Array{FT, 1},
+        u_names::Array{String, 1},
+        scampy_dir::String,
+        scm_names::Array{String, 1},
+        scm_data_root::String,
+    ) where {FT<:AbstractFloat}
+    # create temporary directory to store SCAMPy data in
+    tmpdir = mktempdir(pwd())
+
+    # output directories
+    output_dirs = String[]
+
+    for simname in scm_names
+        # For each scm case, fetch namelist and paramlist
+        inputdir = joinpath(scm_data_root, "Output.$simname.00000")
+        namelist = JSON.parsefile(joinpath(inputdir, "$simname.in"))
+        paramlist = JSON.parsefile(joinpath(inputdir, "paramlist_$simname.in"))
+
+        # update parameter values
+        for (pName, pVal) in zip(u_names, u)
+            paramlist["turbulence"]["EDMF_PrognosticTKE"][pName] = pVal
+        end
+        # write updated paramlist to `tmpdir`
+        paramlist_path = joinpath(tmpdir, "paramlist_$simname.in")
+        open(paramlist_path, "w") do io
+            JSON.print(io, paramlist, 4)
+        end
+
+        # generate random uuid
+        uuid_end = randstring(5)
+        uuid_start = namelist["meta"]["uuid"][1:end-5]
+        namelist["meta"]["uuid"] = "$uuid_start$uuid_end"
+        # set output dir to `tmpdir`
+        namelist["output"]["output_root"] = tmpdir
+        # write updated namelist to `tmpdir`
+        namelist_path = joinpath(tmpdir, "$simname.in")
+        open(namelist_path, "w") do io
+            JSON.print(io, namelist, 4)
+        end
+
+        # run SCAMPy with modified parameters
+        main_path = joinpath(scampy_dir, "main.py")
+        run(`python $main_path $namelist_path $paramlist_path`)
+
+        push!(output_dirs, joinpath(tmpdir, "Output.$simname.$uuid_end"))
```

Comment:
Have you checked whether the generated `Output.$simname.$uuid_end/paramlist_$simname.in` has the correct parameters used for the simulation? I remember I had to copy them back later because SCAMPy just wrote some defaults in the output directory.

Reply:
This was addressed in the most recent PR to SCAMPy ([#104](https://github.com/CliMA/SCAMPy/pull/104)). In particular, see [these lines](https://github.com/CliMA/SCAMPy/commit/496dad0c2438235684823511cacbf5761d6a237c#diff-06280434995cbfb53b907eedba675b41cc005af27569772b446cb08b639c6066R51-R53)